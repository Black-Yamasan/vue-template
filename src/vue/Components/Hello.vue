<template>
  <div class="l-wrap">
    <h2 class="c-heading">{{ msg }}</h2>
    <ul class="c-list">
      <li class="c-list__child"
        v-for="(list, key) in listItem"
        :key="key"
        >List{{list}}</li>
    </ul>
  </div>
</template>

<script>
  export default {
    name: 'Hello',
    data () {
      return {
        msg: 'Hello',
        listItem: [1, 2, 3, 4, 5]
      }
    },
  }
</script>

<style lang="scss" scoped>
  .c-heading {
    margin-bottom: 40px;
    font-size: 2.6rem;
    font-weight: bold;
    letter-spacing: 0.15em;
  }

  .c-list {
    list-style: 'üê±';
    padding-left: 20px;

    &__child {
      margin-bottom: 20px;
      &:last-of-type {
        margin-bottom: 0;
      }
    }
  }
</style>
