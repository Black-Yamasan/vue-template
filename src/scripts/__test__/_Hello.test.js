/**
 * @file _Hello.test
 */
import { shallowMount } from '@vue/test-utils';
import axios from 'axios';
import flushPromises from 'flush-promises';
import Hello from '@/vue/Components/Hello';
jest.mock('axios')

const testListData = [1, 2, 3, 4, 5];
const API_URL = '/assets/data/dammy.json';

describe('Hello Component', () => {
  
  it('set default data', () => {
    expect(typeof Hello.data).toBe('function');
    const defaultData = Hello.data();
    expect(defaultData.msg).toBe('Hello');
  });

  test('get data', () => {
    axios.get.mockImplementation((url) => {
      if ( url === API_URL ) {
        return Promise.resolve({ data: { list: testListData }})
      }
    })
  })

  test('snapshot', async () => {
    const wrapper = shallowMount(Hello, {
      sync: false
    });
  
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  });
});
