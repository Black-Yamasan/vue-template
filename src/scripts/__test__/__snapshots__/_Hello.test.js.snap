// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Component snapshot 1`] = `
<div
  class="l-wrap"
>
  <h2
    class="c-heading"
  >
    Hello
  </h2>
   
  <ul
    class="c-list"
  >
    <list-stub
      class="c-list__child"
      listprops="List1"
    />
    <list-stub
      class="c-list__child"
      listprops="List2"
    />
    <list-stub
      class="c-list__child"
      listprops="List3"
    />
    <list-stub
      class="c-list__child"
      listprops="List4"
    />
    <list-stub
      class="c-list__child"
      listprops="List5"
    />
  </ul>
</div>
`;
